dietpi/dietpi-software

```diff
@@ -10811,6 +10852,73 @@ location = /.well-known/caldav  { return 301 /baikal/html/dav.php; }' > /etc/ngi
 
 		fi
 
+		software_id=125 # Synapse Server
+		if (( ${aSOFTWARE_INSTALL_STATE[$software_id]} == 1 )); then
+
+			Banner_Configuration
+
+			G_EXEC mkdir -p /mnt/dietpi_userdata/synapse
+
+			G_EXEC sudo -u postgres psql -c "create role synapse with login password '$GLOBAL_PW';"
+			G_EXEC sudo -u postgres createdb --encoding=UTF8 --locale=C --template=template0 --owner=synapse synapse
+
+			invalid_text=
+			local servername=
+			local analytics=
+			while :
+			do
+				if G_WHIP_INPUTBOX "${invalid_text}Please enter the name for this homeserver.\nIt is recommended that this be the domain that this server will be hosted on."
+				then
+					servername=$G_WHIP_RETURNED_VALUE
+					break
+				else
+					invalid_text='[ERROR] Please enter a name for your Matrix homserver.\n\n'
+				fi
+			done
+
+			if G_WHIP_YESNO 'Would you like to report basic anonymous analytics, like number of users, to the Matrix foundation?'
+			then
+				analytics='yes'
+			else
+				analytics='no'
+			fi
+
+			G_EXEC cd /mnt/dietpi_userdata/synapse
```

Comment:
Don't forget to `cd /tmp/$G_PROGRAM_NAME` after done. Btw, I think creating and editing the config should only be done if it does not already exist, so that we do not touch or overwrite existing configs?